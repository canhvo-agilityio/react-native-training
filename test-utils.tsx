import { PropsWithChildren, ReactElement } from 'react';
import { render } from '@testing-library/react-native';

const AllTheProviders = ({ children }: PropsWithChildren) => {
  return <>{children}</>;
};

const customRender = (
  ui: ReactElement,
  createNodeMock?: (element: React.ReactElement) => any,
) => render(ui, { wrapper: AllTheProviders, createNodeMock });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
