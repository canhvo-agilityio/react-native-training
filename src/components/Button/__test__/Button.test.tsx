import { Button } from '../index';
import { render, screen } from 'test-utils';

describe('Button', () => {
  it('should render properly', () => {
    const { toJSON } = render(<Button>Press me</Button>);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render correct content in the button', () => {
    render(<Button>Press me</Button>);
    expect(screen.getByText('Press me')).toBeVisible();
  });
});
