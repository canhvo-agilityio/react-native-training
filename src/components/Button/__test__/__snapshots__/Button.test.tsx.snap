// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render properly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "blue",
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "textAlign": "center",
      }
    }
  >
    Press me
  </Text>
</View>
`;
